crates/meilisearch/tests/documents/add_documents.rs

```diff
@@ -18,8 +18,8 @@ async fn add_documents_test_json_content_types() {
         }
     ]);
 
-    // this is a what is expected and should work
-    let server = Server::new().await;
+    // this is what is expected and should work
+    let server = Server::new_shared();
     let app = server.init_web_app().await;
 
     // post
```

Comment:
The shared server is nice but afterward we're hardcoding the name of the index to be dog.
This could be flaky in the future, instead we should create a new unique index.

Also, when we snapshoted the task, we can see it has the task id 0 which is definirely flaky.
We should just call `snapshot!(response, ...);` without the json_string to properly snapshot the tasks